services/repository/src/main/java/org/sagebionetworks/auth/CloudMailInCharacterEncodingFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.sagebionetworks.auth;
+
+import java.io.IOException;
+import java.util.Optional;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletRequestWrapper;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+/**
+ * CloudMailIn does not specify the UTF-8 encoding but may still include UTF-8 characters in its JSON.
+ * We manually override the character encoding to UTF-8 so that Spring can properly handle those characters.
+ */
+public class CloudMailInCharacterEncodingFilter implements Filter {
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+		//nothing to do
+	}
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+		chain.doFilter(new HttpServletRequestWrapper((HttpServletRequest) request) {
+			@Override
+			public String getCharacterEncoding() {
+				return "utf-8";
```

Comment:
Do you want to override the character set in every request or just when it's not specified by the client?

Reply:
We have no idea whether CloudMailIn is specifying a charset or not when they call our API. To be safe and avoid having to wait another 2 weeks for the changes to go in, I always set the charset as utf-8. Since UTF-8 is a superset of both ASCII and ISO-8859-15, there is no risk in having Spring read characters from those charsets as UTF-8. This filter is also restricted to the single CloudMailIn API call so impact is limited.